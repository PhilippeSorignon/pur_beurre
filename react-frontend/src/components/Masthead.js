import React from "react";
import { Link } from 'react-router-dom';

class Masthead extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      search: ""
    }
  }

  render() {
    return (
      <header className="masthead">
        <div className="container h-100">
          <div className="row h-100 align-items-center justify-content-center text-center">
            <div className="col-lg-10 align-self-end">
              <h1 className="text-uppercase text-white font-weight-bold">
                Du gras, oui, mais de qualit√© !
              </h1>
              <hr className="divider my-4" />
            </div>
            <div className="col-lg-8 align-self-baseline">
              <p className="text-white-75 font-weight-light mb-5">
                Trouvez un produit de substitution pour ceux que vous consommez
                tous les jours.
              </p>
              <div className="input-group mb-3">
                <input
                  type="text"
                  className="form-control"
                  placeholder="Produit"
                  aria-label="produit"
                  aria-describedby="button-addon2"
                  onChange={ event => this.setState({ search: event.target.value }) }
                />
                <div className="input-group-append">
                  <Link
                    to={`search/${this.state.search}`}
                    className="btn btn-outline-secondary"
                    role="button"
                    type="button"
                    id="button-addon2"
                  >
                    Chercher
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
    );
  }
}

export default Masthead;
