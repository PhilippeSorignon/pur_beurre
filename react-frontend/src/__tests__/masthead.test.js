import React from 'react';
import ReactDOM from 'react-dom';
import Masthead from '../components/Masthead';

import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Masthead renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Masthead />, div);
});

it("Test if the link update on user entry", () => {
  const { getByLabelText, getByText } = render(<Masthead />);
  const searchInput = getByLabelText('produit');

  fireEvent.change(searchInput, { target: { value: "nutella" } })

  expect(getByText('Chercher')).toHaveAttribute("href", "search/nutella");
});

it("Masthead matches snapshot", () => {
  const tree = renderer.create(<Masthead />).toJSON();
  expect(tree).toMatchSnapshot();
})
