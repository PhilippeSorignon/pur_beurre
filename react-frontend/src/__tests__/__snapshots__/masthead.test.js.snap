// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masthead matches snapshot 1`] = `
<header
  className="masthead"
>
  <div
    className="container h-100"
  >
    <div
      className="row h-100 align-items-center justify-content-center text-center"
    >
      <div
        className="col-lg-10 align-self-end"
      >
        <h1
          className="text-uppercase text-white font-weight-bold"
        >
          Du gras, oui, mais de qualit√© !
        </h1>
        <hr
          className="divider my-4"
        />
      </div>
      <div
        className="col-lg-8 align-self-baseline"
      >
        <p
          className="text-white-75 font-weight-light mb-5"
        >
          Trouvez un produit de substitution pour ceux que vous consommez tous les jours.
        </p>
        <div
          className="input-group mb-3"
        >
          <input
            aria-describedby="button-addon2"
            aria-label="produit"
            className="form-control"
            onChange={[Function]}
            placeholder="Produit"
            type="text"
          />
          <div
            className="input-group-append"
          >
            <a
              className="btn btn-outline-secondary"
              href="search/"
              id="button-addon2"
              role="button"
              type="button"
            >
              Chercher
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
